src/session_impl.cpp

```diff
@@ -1887,6 +1864,50 @@ namespace {
 		}
 	}
 
+	void session_impl::remove_listen_sockets(std::vector<listen_endpoint_t>& eps)
+	{
+		m_listen_sockets.remove_if([&eps, this](listen_socket_t const& sock)
```

Comment:
oh, I am. this is a ``std::list``, and ``remove_if``-member function means what you think it does (as opposed to the free function ``remove_if``). never mind